File: Practice(Beginner)/C++/Similar Dishes.cpp
// Source: https://www.codechef.com/problems/SIMDISH

#include <bits/stdc++.h>
using namespace std;
int main(){
    int t, temp;
    cin >> t;
    while (t--){
        vector<string > line1;
        vector<string > line2;
        string s;
        for (int i = 0; i < 4; ++i) {
            cin >> s;
            line1.push_back(s);
        }
        for (int i = 0; i < 4; ++i) {
            cin >> s;
            line2.push_back(s);
        }
        int c = 0;
        bool flag = false;
        for (int i = 0; i < 4; ++i) {
            auto it = find(line2.begin(),line2.end(),line1[i]);
            if(it != line2.end()){
                c++;
            }
            if(c == 2){
                flag = true;
                break;
            }
        }
        flag ? cout << "similar" << endl : cout << "dissimilar" << endl ;
    }
}

File: Practice(Beginner)/Python3/Similar_Dishes.py
# Source: https://www.codechef.com/problems/SIMDISH

t = int(input())
for test in range(t):
    line1 = list(map(str, input().split()))
    line2 = list(map(str, input().split()))
    c = 0
    flag = False
    for i in line1:
        if(i in line2):
            c += 1
        if(c == 2):
            flag =True
            break
    print("similar") if(flag) else print("dissimilar")